app/utils/useOpenMap.tsx

```diff
@@ -0,0 +1,23 @@
+import { useCallback } from "react"
+import { Linking, Platform } from "react-native"
+
+export const useOpenMap = () => {
+  const openMap = useCallback(
+    async (address: string, zipCode: string, city: string) => {
+      const destination = encodeURIComponent(`${address} ${zipCode}, ${city}`)
+      const provider = Platform.OS === "ios" ? "apple" : "google"
+      const link = `http://maps.${provider}.com/?daddr=${destination}`
+
+      try {
+        const supported = await Linking.canOpenURL(link)
+
+        if (supported) Linking.openURL(link)
+      } catch (error) {
+        console.log(error)
+      }
+    },
+    []
+  )
+
+  return openMap
+}
```

Comment:
Since this function doesn't use any hooks other than `useCallback`, could it just be a normal function rather than a quasi-hook? `useCallback` seems unnecessary here, as all we're doing is defining a function and opening a URL with that function.

Reply:
e.g., remove the `useOpenMap` wrapper and just export the function `openMap` directly from this file.

Reply:
agreed 👍